src/components/smart_objects/test/SmartObjectUnit_test.cc

```diff
@@ -585,6 +586,26 @@ TEST(MapEraseTest, SmartObjectTest) {
 }
 // TODO: Add a test to check accessing an array at strange indexes.
 
+TEST(DoubleCleanupDataTest, SmartObjectTest) {
+  class DerivedSmartObject : public SmartObject {
+   public:
+    DerivedSmartObject(SmartType Type) : SmartObject(Type) {}
+    using SmartObject::operator=;
+    void cleanup_data() {
+      SmartObject::cleanup_data();
+    }
+  };
+
+  DerivedSmartObject obj(SmartType_String);
+  ASSERT_EQ(SmartType_String, obj.getType());
+
+  obj = "test string_value";
+  ASSERT_EQ(std::string("test string_value"), obj.asString());
+
+  obj.cleanup_data();
+  obj.cleanup_data();
```

Comment:
@SNiukalov It's confusing...Why did you call it twice?

Reply:
@AByzhynar 
In this case we try testing stability work clean_data function.
In previously implementation we will get crash or indefine behevior after twice call this function.
Unit tests will be finished with aborting.
But now this test case will working correctly.

Fixed in:https://github.com/smartdevicelink/sdl_core/pull/3146/commits/6915e49d17c1a2bb31e654ab1fa2fc3b668b9040